wireless_watcher/scripts/watcher_node

```diff
@@ -70,17 +72,34 @@ def main():
     rospy.init_node('wireless_watcher')
 
     hz = rospy.get_param('~hz', 1)
-
     dev = rospy.get_param('~dev', None)
-    if not dev:
+    robot = rospy.get_param('/robot_name', 'p3-001')
+    chip = rospy.get_param('~wifi_hardware', None)
+    firmware_version = rospy.get_param("~wifi_firmware", None)
+    if dev is None:
         wldevs = [d for d in os.listdir(SYS_NET_PATH) if d.startswith('wl')]
         if wldevs:
             dev = wldevs[0]
         else:
             rospy.logfatal("No wireless device found to monitor.")
             return 1
     rospy.loginfo("Monitoring %s" % dev)
-
+    if firmware_version is None:
+        firmware_data = grep(subprocess.check_output(["ethtool", "-i", dev], stderr=subprocess.STDOUT).split("\n"),
+                             "firmware")
+        if len(firmware_data) == 1:
+            firmware_version = firmware_data[0].split(' ')[1]
+        else:
+            firmware_version = "unknown"
+            rospy.logwarn("Unable to detect wifi hardware firmware version. Using {}".format(firmware_version))
+    if chip is None:
+        devices = grep(subprocess.check_output(["lspci"], stderr=subprocess.STDOUT).split("\n"), "Network controller")
+        if len(devices) == 1:
+            chip = devices[0].split(":")[2][1:]
+        else:
+            chip = "unknown"
+            rospy.logwarn("Unable to detect wifi hardware chip version. Using {}".format(chip))
```

Comment:
I wonder if it would be useful to always log what chip and firmware version being used.